pac4j-core/src/main/java/org/pac4j/core/profile/ProfileManager.java

```diff
@@ -34,19 +34,16 @@ public ProfileManager(final WebContext context) {
     }
 
     /**
-     * Retrieve the current user profile.
+     * Retrieve the current user profile (from request first and then from the session if not found and requested).
      *
      * @param readFromSession if the user profile must be read from session
      * @return the user profile
      */
     public U get(final boolean readFromSession) {
-        U profile = null;
-        if (readFromSession) {
+        U profile = (U) this.context.getRequestAttribute(Pac4jConstants.USER_PROFILE);
```

Comment:
Notice that now, the user profile is first retrieved from the request before being potentially retrieved from the session: in that case, a REST authentication (by JWT token) can override an authentication from an indirect client (like Facebook) if the session is not used by the direct client.